### PR TITLE
DOC-2352: On-premises `Import from Word and Export to Word` documentation.

### PR DESCRIPTION
**Pull request by kemister85**
Ticket: DOC-2352

Site: [DOC-2339_DOC-2352 site](http://docs-hotfix-7-doc-2339doc-2352.staging.tiny.cloud/docs/tinymce/latest/individual-import-from-word-and-export-to-word-on-premises/)

Changes:
* Add On-premises `Import from Word and Export to Word` documentation.

Pre-checks:
- [x] Branch prefixed with `feature/7/` or `hotfix/7/`

Review:
- [x] Documentation Team Lead has reviewed

### CODE REVIEW

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-overview-on-premises.adoc [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+{pluginname} is a set of two converter services that allow for both exporting a structured HTML content (e.g. created with {productname} WYSIWYG editor) into a `.docx` Microsoft Word file and for importing content from `.docx` and `.dotx` files and converting it into a styled HTML document with comments and suggestions attached.
+
+About xref:importword.adoc[Import from Word] and xref:exportword.adoc[Export to Word] features you can read in a dedicated sections of the documentation.
+
+The On-Premises version of the {pluginname} features is an application that can be installed and run on the customer’s in-house servers and computing infrastructure, including a private cloud. It contains all the features of the {pluginname} converters available as SaaS.
+
+A valid license key is needed in order to install {pluginname} On-Premises.
+
+link:https://www.tiny.cloud/contact/[Contact us] for a trial license key.
```
Comment (ShiridiGandham) [inline modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-overview-on-premises.adoc:12]:
```suggestion
A valid license key is needed in order to install {pluginname} On-Premises. link:https://www.tiny.cloud/contact/[Contact us] for a trial license key.
```

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-installation-on-premises.adoc [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+==== Containerize example using docker
+
+Launch the Docker container:
+
+[source, sh, subs="attributes+"]
+----
+docker run --init -p 8080:8080 -e LICENSE_KEY=[your_license_key] {dockerimageimportfromwordexporttoword}:[version]
+----
+
```
Comment (ShiridiGandham) [inline modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-installation-on-premises.adoc:29]:
```suggestion

Login to Docker registry:

[source, sh, subs="attributes+"]
----
docker login -u [username] -p [password] registry.containers.tiny.cloud
----

Launch the Docker container:

[source, sh, subs="attributes+"]
----
docker run --init -p 8080:8080 -e LICENSE_KEY=[your_license_key] {dockerimageimportfromwordexporttoword}:[version]
----

```

File: antora.yml [hunk 3]
```diff
@@ -16,6 +16,8 @@ asciidoc:
     webcomponent_url: https://cdn.jsdelivr.net/npm/@tinymce/tinymce-webcomponent@2/dist/tinymce-webcomponent.min.js
     jquery_url: https://cdn.jsdelivr.net/npm/@tinymce/tinymce-jquery@2/dist/tinymce-jquery.min.js
     default_meta_keywords: tinymce, documentation, docs, plugins, customizable skins, configuration, examples, html, php, java, javascript, image editor, inline editor, distraction-free editor, classic editor, wysiwyg
+    # product docker variables
+    dockerimageimportfromwordexporttoword: registry.containers.tiny.cloud/import-from-word-export-to-word
```
Comment (ShiridiGandham) [inline antora.yml:20]:
```suggestion
    # product docker variables
    dockerimageimportfromwordexporttoword: registry.containers.tiny.cloud/docx-converter-tiny
```

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-installation-on-premises.adoc [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    image: {dockerimageimportfromwordexporttoword}:[version]
+      ports:
+        - "8080:8080"
+      restart: always
+      init: true
+      environment:
+        LICENSE_KEY: your_license_key
+        # Secret Key is optional
+        SECRET_KEY: secret_key
+----
```
Comment (ShiridiGandham) [inline modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-installation-on-premises.adoc:56]:
```suggestion
----
version: "3.8"
services:
  doc-converter:
    image: {dockerimageimportfromwordexporttoword}:[version]
    ports:
      - "8080:8080"
    restart: always
    init: true
    environment:
      LICENSE_KEY: "licensekey"
      # Secret Key is optional
      SECRET_KEY: "secret_key"
----
```

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-overview-on-premises.adoc [hunk 5]
```diff
@@ -0,0 +1,14 @@
+[[overview]]
+== Overview
+
+{pluginname} is a set of two converter services that allow for both exporting a structured HTML content (e.g. created with {productname} WYSIWYG editor) into a `.docx` Microsoft Word file and for importing content from `.docx` and `.dotx` files and converting it into a styled HTML document with comments and suggestions attached.
```
Comment (lostkeys) [inline modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-overview-on-premises.adoc:4]: Remove " with comments and suggestions attached"

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-installation-on-premises.adoc [hunk 6]
```diff
@@ -0,0 +1,88 @@
+[[installation]]
+== Installation
+
+> A valid license key is needed in order to install {pluginname} On-Premises.
```
Comment (lostkeys) [inline modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-installation-on-premises.adoc:4]: Convert to a note-box

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-installation-on-premises.adoc [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+The application is provided as a docker image by default.
+
+It can be run with any Open Container runtime tool e.g. link:https://kubernetes.io/[Kubernetes], link:https://www.redhat.com/en/technologies/cloud-computing/openshift[OpenShift], link:https://podman.io/[Podman], link:https://docs.docker.com/[Docker] and many others.
+
+Refer to the xref:individual-import-from-word-and-export-to-word-on-premises.adoc#requirements[Requirements guide] for more information about the hardware and software requirements to run the {pluginname} On-Premises.
+
+=== Setting up the application using a Docker container
+
+. Use instructions from the {productname} Ecosystem customer dashboard to log into the {pluginname} On-Premises (PDF converter) Docker registry and pull the Docker image.
```
Comment (lostkeys) [inline modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-installation-on-premises.adoc:17]: Change to something like

> Use the username and password provided to you by Tiny to login to the Docker registry and pull the Docker image.

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-autorization-on-premises.adoc [hunk 8]
```diff
@@ -0,0 +1,68 @@
+[[authorization]]
+== Authorization
+
+To enable authorization, set the `SECRET_KEY` environment variable during the xref:individual-import-from-word-and-export-to-word-on-premises.adoc#installation[installation].
+
+If the `SECRET_KEY` variable is set, then all requests must have a header with a JWT (JSON Web Token) signed with this key. The token should be passed as a value of the `Authorization` header for each request sent to the {pluginname} REST API.
+
+> If the `SECRET_KEY` is not setup during the installation, then {pluginname} On-Premises will not require any headers with tokens when sending requests to the {pluginname} REST API. However, this it is not recommend to skip the authorization when running {pluginname} On-Premises in a public network.
```
Comment (lostkeys) [inline modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-autorization-on-premises.adoc:8]: Convert to a note-box

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-api-usage-on-premises.adoc [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+== API usage
+
+{pluginname} On-Premises consists of two services:
+
+* xref:exportword.adoc[Export to Word] provides conversion from an HTML document to a `.docx` file via Restful API.
+* xref:importword.adoc[Import from Word] allows importing `.docx` file and converting it into a styled HTML document with comments and suggestions attached.
+
+The API is available on `http://localhost:[port]` (by default the port is 8080).
+
+> The REST API documentation is available at `http://localhost:[port]/docs`.
```
Comment (lostkeys) [inline modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-api-usage-on-premises.adoc:11]: Convert to a note-box

Comment (lostkeys) [inline modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-api-usage-on-premises.adoc:11]: The URL does not render properly on the site. It should read `http://localhost:[port]/docs`.

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-logs-on-premises.adoc [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+See example charts to check how to use logs for monitoring purposes
+////
+
+=== Docker
+
+The docker has built-in logging mechanisms that capture logs from the output of the containers. The default logging driver writes the logs to files.
+
+When using this driver, use the `docker logs` command to show logs from a container. Use the `-f` flag to view logs in real time. Refer to the link:https://docs.docker.com/engine/reference/commandline/logs/[official Docker documentation] for more information about the logs command.
+
+> When a container is running for a long period of time, the logs can take up a lot of space. To avoid this problem, you should make sure that the log rotation is enabled. This can be set with the `max-size` option.
```
Comment (lostkeys) [inline modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-logs-on-premises.adoc:61]: Convert to a note-box

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-logs-on-premises.adoc [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+* Google Cloud: link:https://cloud.google.com/logging[Cloud Logging^]
+* Azure: link:https://azure.microsoft.com/en-us/services/monitor/[Azure Monitor^]
+
+To use CloudWatch with AWS ECS, a log group must be created before, and the log driver must be changed to `awslogs`. When the log driver is configured properly, logs will be streamed directly to CloudWatch.
+
+The `logConfiguration` may look similar to this:
+
+[source, json]
+----
+"logConfiguration": {
```
Comment (lostkeys) [inline modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-logs-on-premises.adoc:83]: This block contains references to our friends that needs to be changed.

**Review by lostkeys — CHANGES_REQUESTED** [review]

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-logs-on-premises.adoc [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+  "logDriver": "awslogs",
+  "options": {
+    "awslogs-region": "us-west-2",
+    "awslogs-group": "cksource",
+    "awslogs-stream-prefix": "ck-docx-converter-logs"
+  }
+}
+----
```
Comment (kemister85) [inline modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-logs-on-premises.adoc:91]:
```suggestion
The `logConfiguration` may look similar to this:

[source, json]
----
"logConfiguration": {
  "logDriver": "awslogs",
  "options": {
    "awslogs-region": "us-west-2",
    "awslogs-group": "tinysource",
    "awslogs-stream-prefix": "tiny-docx-converter-logs"
  }
}
----
```

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-logs-on-premises.adoc [hunk 13]
```diff
@@ -0,0 +1,210 @@
...
+"logConfiguration": {
+  "logDriver": "awslogs",
+  "options": {
+    "awslogs-region": "us-west-2",
+    "awslogs-group": "tinysource",
+    "awslogs-stream-prefix": "tiny-docx-converter-logs"
+  }
+}
+----
```
Comment (kemister85) [inline modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-logs-on-premises.adoc:91]: @ShiridiGandham we need to confirm this before we merge.

```suggestion
The `logConfiguration` may look similar to this:

[source, json]
----
"logConfiguration": {
  "logDriver": "awslogs",
  "options": {
    "awslogs-region": "us-west-2",
    "awslogs-group": "tinysource",
    "awslogs-stream-prefix": "tiny-docx-converter-logs"
  }
}
----
```

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-overview-on-premises.adoc (hunk 5 above)
Comment (kemister85) [reply modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-overview-on-premises.adoc:4]:
```suggestion
{pluginname} is a set of two converter services that allow for both exporting a structured HTML content (e.g. created with {productname} WYSIWYG editor) into a `.docx` Microsoft Word file and for importing content from `.docx` and `.dotx` files and converting it into a styled HTML document.
```

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-overview-on-premises.adoc [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+
+About xref:importword.adoc[Import from Word] and xref:exportword.adoc[Export to Word] features you can read in a dedicated sections of the documentation.
+
+The On-Premises version of the {pluginname} features is an application that can be installed and run on the customer’s in-house servers and computing infrastructure, including a private cloud. It contains all the features of the {pluginname} converters available as SaaS.
```
[the remark this answers is not in the document]
Comment (kemister85) [reply modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-overview-on-premises.adoc:8]:
```suggestion
```

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-installation-on-premises.adoc (hunk 6 above)
Comment (kemister85) [reply modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-installation-on-premises.adoc:4]:
```suggestion
[NOTE]
A valid license key is needed in order to install {pluginname} On-Premises.
```

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-overview-on-premises.adoc [hunk 15]
```diff
@@ -0,0 +1,14 @@
+[[overview]]
+== Overview
+
+{pluginname} is a set of two converter services that allow for both exporting a structured HTML content (e.g. created with {productname} WYSIWYG editor) into a `.docx` Microsoft Word file and for importing content from `.docx` and `.dotx` files and converting it into a styled HTML document with comments and suggestions attached.
+
+About xref:importword.adoc[Import from Word] and xref:exportword.adoc[Export to Word] features you can read in a dedicated sections of the documentation.
```
[the remark this answers is not in the document]
Comment (kemister85) [reply modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-overview-on-premises.adoc:6]:
```suggestion
See xref:importword.adoc[Import from Word] and xref:exportword.adoc[Export to Word] featured documentation for more information on these plugins.
```

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-api-usage-on-premises.adoc (hunk 9 above)
Comment (kemister85) [reply modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-api-usage-on-premises.adoc:11]:
```suggestion
[NOTE]
The REST API documentation is available at `+http://localhost:[port]/docs+`.
```

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-api-usage-on-premises.adoc [hunk 16]
```diff
@@ -0,0 +1,14 @@
+[[api-usage]]
+== API usage
+
+{pluginname} On-Premises consists of two services:
+
+* xref:exportword.adoc[Export to Word] provides conversion from an HTML document to a `.docx` file via Restful API.
+* xref:importword.adoc[Import from Word] allows importing `.docx` file and converting it into a styled HTML document with comments and suggestions attached.
+
+The API is available on `http://localhost:[port]` (by default the port is 8080).
```
[the remark this answers is not in the document]
Comment (kemister85) [reply modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-api-usage-on-premises.adoc:9]:
```suggestion
The API is available on `+http://localhost:[port]+` (by default the port is 8080).
```

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-autorization-on-premises.adoc (hunk 8 above)
Comment (kemister85) [reply modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-autorization-on-premises.adoc:8]:
```suggestion
[NOTE]
If the `SECRET_KEY` is not setup during the installation, then {pluginname} On-Premises will not require any headers with tokens when sending requests to the {pluginname} REST API. However, this it is not recommend to skip the authorization when running {pluginname} On-Premises in a public network.
```

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-autorization-on-premises.adoc [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+
+=== Generating the token
+
+{productname} recommends using the libraries listed on link:http://jwt.io/[jwt.io] to generate the token. The token is considered valid, when:
```
[the remark this answers is not in the document]
Comment (kemister85) [reply modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-autorization-on-premises.adoc:13]:
```suggestion
{companyname} recommends using the libraries listed on link:http://jwt.io/[jwt.io] to generate the token. The token is considered valid, when:
```

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-overview-on-premises.adoc (hunk 15 above)
[the remark this answers is not in the document]
Comment (kemister85) [reply modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-overview-on-premises.adoc:6]:
```suggestion
```

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-installation-on-premises.adoc [hunk 18]
```diff
@@ -0,0 +1,89 @@
+[[installation]]
+== Installation
+
+[NOTE]
+A valid license key is needed in order to install {pluginname} On-Premises.
+link:https://www.tiny.cloud/contact/[Contact us] for a trial license key.
```
Comment (kemister85) [inline modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-installation-on-premises.adoc:6]: @lostkeys we should remove this if we do not offer a `trail license key`.
```suggestion
```

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-logs-on-premises.adoc (hunk 13 above)
Comment (kemister85) [reply modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-logs-on-premises.adoc:89]: SysDel have confirmed this is adequate if we leave as is.

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-overview-on-premises.adoc [hunk 19]
```diff
@@ -0,0 +1,12 @@
+[[overview]]
+== Overview
+
+{pluginname} is a set of two converter services that allow for both exporting a structured HTML content (e.g. created with {productname} WYSIWYG editor) into a `.docx` Microsoft Word file and for importing content from `.docx` and `.dotx` files and converting it into a styled HTML document.
+
+
```
Comment (kemister85) [inline modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-overview-on-premises.adoc:6]:
```suggestion
```

File: modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-overview-on-premises.adoc [hunk 20]
```diff
@@ -0,0 +1,10 @@
+[[overview]]
+== Overview
+
+{pluginname} is a set of two converter services that allow for both exporting a structured HTML content (e.g. created with {productname} WYSIWYG editor) into a `.docx` Microsoft Word file and for importing content from `.docx` and `.dotx` files and converting it into a styled HTML document.
+
+A valid license key is needed in order to install {pluginname} On-Premises. link:https://www.tiny.cloud/contact/[Contact us] for a trial license key.
```
Comment (kemister85) [inline modules/ROOT/partials/individually-licensed-components/import-from-word-and-export-to-word/import-from-word-and-export-to-word-overview-on-premises.adoc:6]: @lostkeys if we do not offer a `trail licence key` then we should remove this.
```suggestion

A valid license key is needed in order to install {pluginname} On-Premises.
```

**Review by ShiridiGandham — APPROVED** [review]
LGTM.

**Review by MitchC1999 — APPROVED** [review]

**Pull request merged**